src/utils/paths.js

```diff
@@ -41,6 +41,32 @@ export class PathConditionsImplementation extends PathConditions {
     this._failedNegativeImplications = undefined;
   }
 
+  // It makes the strong assumption that, in order for 2 path conditions to be equal,
```

Comment:
I think you could make this more general by treating two sets of path conditions equivalent if set a implies all of the conditions in set b and vice versa. I.e. a <=> b is as good as a = b.